db/migrate/20240731175623_add_course_type_to_courses.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddCourseTypeToCourses < ActiveRecord::Migration[7.1]
+  def change
+    add_column :course, :course_type, :integer, default: 0, null: false
```

Comment:
How do you feel about using a `string` here. The database will be easier to read `postgraduate` rather than `0`, `postgraduate` and `undergraduate` values are unlikely to change

Reply:
Should we add an index for this field? *I am not sure how will gonna be used so I didn't suggested but worth a thinking.

Reply:
[I've converted it to a string](https://github.com/DFE-Digital/publish-teacher-training/pull/4415/commits/d4f054f17748643d10ae86d5e4dc5eaadf7fefee) @dcyoung-dev. I mentioned in my PR description that I don't feel strongly either way. Using strings makes reading the database by a non-dev slightly easier, but mapping the integers isn't that difficult. Using integers offers very slight performance gains, but it's probably not significant enough to matter in our case, however that's what was done previously and most recently for `application_status`.

@tomas-stefano, I don't think we need an index. We won't be querying by `course_type` very often, if at all. The only courses that aren't postgraduate are TDA courses, and we query those by `program_type`. Adding an index would make updates to the column slower without much benefit for reading the column, since we won't be reading it frequently.

But I'm open to hearing any counterarguments if anyone thinks there's a strong case for it.

Reply:
Ok, makes sense. For what I understand we will only need a index in Apply because we will be filtering in Manage.